packages/graph/hash_graph/lib/graph/src/ontology/property_type.rs

```diff
@@ -26,7 +31,243 @@ impl QueryRecord for PropertyType {
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PropertyTypeQueryToken {
+    OwnedById,
+    BaseUri,
+    VersionedUri,
+    Version,
+    Title,
+    Description,
+    DataTypes,
+    PropertyTypes,
+}
+
+/// Deserializes a [`PropertyTypeQueryPath`] from a string sequence.
+pub struct PropertyTypeQueryPathVisitor {
+    /// The current position in the sequence when deserializing.
+    position: usize,
+}
+
+impl PropertyTypeQueryPathVisitor {
+    #[must_use]
+    pub const fn new(position: usize) -> Self {
+        Self { position }
+    }
+}
+
+impl<'de> Visitor<'de> for PropertyTypeQueryPathVisitor {
+    type Value = PropertyTypeQueryPath<'de>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str(
+            "one of `ownedById`, `baseUri`, `versionedUri`, `version`, `title, `description`, \
+             `dataTypes`, or `propertyTypes`",
+        )
+    }
+
+    fn visit_seq<A>(mut self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let token = seq
+            .next_element()?
+            .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+        self.position += 1;
+        Ok(match token {
+            PropertyTypeQueryToken::OwnedById => PropertyTypeQueryPath::OwnedById,
+            PropertyTypeQueryToken::BaseUri => PropertyTypeQueryPath::BaseUri,
+            PropertyTypeQueryToken::VersionedUri => PropertyTypeQueryPath::VersionedUri,
+            PropertyTypeQueryToken::Version => PropertyTypeQueryPath::Version,
+            PropertyTypeQueryToken::Title => PropertyTypeQueryPath::Title,
+            PropertyTypeQueryToken::Description => PropertyTypeQueryPath::Description,
+            PropertyTypeQueryToken::DataTypes => {
+                seq.next_element::<Selector>()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                self.position += 1;
+
+                let data_type_query_path =
+                    DataTypeQueryPathVisitor::new(self.position).visit_seq(seq)?;
+
+                PropertyTypeQueryPath::DataTypes(data_type_query_path)
+            }
+            PropertyTypeQueryToken::PropertyTypes => {
+                seq.next_element::<Selector>()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                self.position += 1;
+
+                let property_type_query_path = Self::new(self.position).visit_seq(seq)?;
+
+                PropertyTypeQueryPath::PropertyTypes(Box::new(property_type_query_path))
+            }
+        })
+    }
+}
+impl<'de: 'k, 'k> Deserialize<'de> for PropertyTypeQueryPath<'k> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(PropertyTypeQueryPathVisitor::new(0))
+    }
+}
+
+pub enum Selector {
+    Wildcard,
```

Comment:
You've explicitly called this an asterisk elsewhere right? Is there a reason for using different terms in the various contexts?

Reply:
I used `Wildcard` before I used the term `Asterisk`. I changed the name in a377aa76ad3b9865e0dfe4cda25dc44f056a2300